lib/audited/auditor.rb

```diff
@@ -220,30 +222,21 @@ def write_audit(attrs)
       end
 
       def presence_of_audit_comment
-        case
-        when !auditing_enabled
-          return true
-        when audit_comment.present?
-          return true
-        when audited_options[:on].exclude?(:create) && self.new_record?
-          return true
-        when audited_options[:on].exclude?(:update) && self.persisted?
-          return true
-        when audited_changes.empty? && self.persisted?
-          return true
-        else
-          errors.add(:audit_comment, "can't be blank.")
-        end
+        require_comment if comment_required_state?
       end
 
       def require_comment
         if auditing_enabled && audit_comment.blank?
```

Comment:
So, in the case of callbacks, if this is getting called by `before_create`, `before_update`, etc. then all we need to check are these two things. If these criteria aren't met, then the state is invalid and it won't be created/updated/destroyed.